File: tests/fixtures/go/basics.go
package main

import (
    "fmt"
    "net/http"
)

type User struct {
    Name string `json:"name"`
    Age  int    `json:"age"`
}

func (u *User) Greet() string {
    return fmt.Sprintf("Hello, %s!", u.Name)
}

func main() {
    user := &User{Name: "John", Age: 30}
    fmt.Println(user.Greet())
    
    for i := 0; i < 10; i++ {
        fmt.Println(i)
    }
    
    switch user.Age {
    case 0:
        fmt.Println("newborn")
    default:
        fmt.Println("older")
    }
}

File: tests/fixtures/php/basics.php
<?php
namespace App\Models;

use App\Traits\Loggable;

class User {
    public string $name;
    private int $age;
    
    public function __construct(string $name, int $age) {
        $this->name = $name;
        $this->age = $age;
    }
    
    public function greet(): string {
        return "Hello, {$this->name}!";
    }
}

$user = new User("John", 30);
echo $user->greet();

// Arrow function
$double = fn($x) => $x * 2;

// Match expression
$result = match($status) {
    "active" => 1,
    "inactive" => 0,
    default => -1,
};
?>
